src/briefcase/commands/create.py

```diff
@@ -513,12 +513,19 @@ def _write_requirements_file(
             written.
         """
 
+        extra_pip_args = self._extra_pip_args(app)
+
         with self.input.wait_bar("Writing requirements file..."):
             with requirements_path.open("w", encoding="utf-8") as f:
-                if requires:
+                if requires or extra_pip_args:
```

Comment:
My inclination is that this is the right idea, not taken far enough. If we're generating a new file, we should *always* be writing the timestamp - worst case, we end up with a file that only contains the timestamp, rather than a literally empty file.

Reply:
Totally, makes sense to do away with the check altogether.

Also noting that I'd originally moved the check to the function root and done an early return when nothing would be written, but if requirements are being removed, it still needs to write the file to reflect requirements changed. Removing the conditional on the timestamp write makes that aspect a bit clearer as well (maybe with a small comment to help explain).